src/main/java/seedu/address/model/FocusMode.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents the state of the focus mode.
+ */
+public class FocusMode implements ReadOnlyFocusMode {
+
+    private boolean isActive;
```

Comment:
Is the `isActive` variable necessary? I think it is recoverable from `focusedClass != null`. So this can be removed.